bin/install

```diff
@@ -33,48 +54,52 @@ install_deno() {
   local uncompress_command
   local archive_file
 
-  if [[ $version > "0.35.0" ]]; then
-    case "$OSTYPE" in
-      darwin*) platform="apple-darwin" ;;
-      linux*) platform="unknown-linux-gnu" ;;
-      *) fail "Unsupported platform" ;;
-    esac
-
-    case "$(uname -m)" in
-      x86_64) architecture="x86_64" ;;
-      arm64) architecture="aarch64" ;;
-      *) fail "Unsupported architecture" ;;
-    esac
-
-    archive_format="zip"
-    archive_file="deno-${architecture}-${platform}.${archive_format}"
-    uncompress_command="unzip -d ${install_path}/bin"
+  if [ "$install_type" == "version" ]; then
```

Comment:
It's only possible to install from source if `ASDF_INSTALL_TYPE` is `ref`, in which case the appropriate variable name inside the function might be `ref`, not `version`. Going further, I don't know if cargo supports installation using ref. Are you sure about that?
https://asdf-vm.com/plugins/create.html#environment-variables

Reply:
Cargo did support install through git ref/tag/branch, see (install options)[https://doc.rust-lang.org/cargo/commands/cargo-install.html#install-options] for the flags. Now, this the case of `ref`, this `ref` that get passed from ASDF to the plugin is coming from the version user specified in `asdf install`. So if they specified `asdf install deno ref:latest`, `ref` will have a value of `latest`.

So for the changes, I think I will
- Rename the variable name in the `install_from_source` function to be `ref` instead, as it could be a git commit sha
- Add a basic git sha check (check the length of the string to match specific length), and use that as a `--rev` in the Cargo command
- Add documentation on how to install specific version/commit from the source